### PR TITLE
Electron debugger

### PR DESCRIPTION
**Pull request by christianalfoni**
Allright this is pretty awesome!

- I have structured pretty well now, so building makes sense:

`npm run build:electron` or `npm run build:chromeExtension`

Both of these uses the same `src` folder to build the debugger itself. The webpack config decides which `connector` to use.

Electron application is outputted in: `cerebral-debugger/electron/dist`, the Chrome extension is in `cerebral-debugger/chromeExtension`

- To develop you use `npm start`, which uses electron. This is great, because now we can develop the debugger using a real app to test handling of the passed data. This was a pain earlier, no need to simulate anymore

To try it out just add config:

```js
Controller({
  devtools: Devtools({
    remoteDebugger: 'localhost:8080'
  })
})
```

:)

### CODE REVIEW

**Comment by christianalfoni** [comment]
Okay, I updated the scripts and cleaned things up. I think this is ready for a merge. Not quite sure how we should distribute the Electron app though... if we should zip locally and upload somewhere? Or just push it to github?

File: package.json [hunk 1]
```diff
@@ -52,7 +52,8 @@
     "ignore": [
       "**/node_modules/",
       "**/build/",
-      "packages/cerebral-debugger/public/prism.js"
+      "packages/cerebral-debugger/chromeExtension/prism.js",
+      "packages/cerebral-debugger/electron/prism.js"
```
Comment (Guria) [inline package.json]: could we require prism from npm?

File: packages/cerebral-debugger/electron/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+CC0 1.0 Universal
```
Comment (Guria) [inline packages/cerebral-debugger/electron/LICENSE.md]: Is it intensional or just copy paste?

File: packages/cerebral-debugger/package.json [hunk 3]
```diff
@@ -1,38 +1,60 @@
 {
-  "name": "cerebral-debugger",
-  "version": "2.0.0",
+  "name": "electron-quick-start",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: better keep cerebral-debugger here

File: packages/cerebral-debugger/package.json [hunk 4]
```diff
@@ -1,38 +1,60 @@
...
+  "version": "1.0.0",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: and version too :)

File: packages/cerebral-debugger/package.json [hunk 5]
```diff
@@ -1,38 +1,60 @@
...
   "private": true,
-  "description": "Cerebral Debugger",
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/cerebral/cerebral.git"
+  "description": "A minimal Electron application",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: the same about description :)
or we can omit it for private packages I think

File: packages/cerebral-debugger/package.json [hunk 6]
```diff
@@ -1,38 +1,60 @@
...
+  "main": "electron/main.js",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: should we have this main here or one from electron/package.json should be enough?

File: packages/cerebral-debugger/package.json [hunk 7]
```diff
@@ -1,38 +1,60 @@
...
+  "scripts": {
+    "start": "DEBUGGER_TARGET=electron NODE_ENV=development concurrently -k 'node ./electron/devServer.js' 'npm run electron'",
+    "electron": "electron .",
+    "build:chromeExtension": "NODE_ENV=production webpack --config webpack.chromeExtension.config",
+    "build:electron": "NODE_ENV=production webpack --config webpack.electron.config && npm run prepare:electron && npm run package:electron",
+    "prepare:electron": "rimraf electron/dist electron/node_modules && copyfiles 'node_modules/ws/**/*' 'node_modules/ultron/**/*' 'node_modules/options/**/*' electron",
+    "package:electron": "electron-packager ./electron CerebralDebugger --platform=darwin --out ./electron/dist"
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: Why only darwin? btw we should explore if we can build for all platforms on travis.

File: packages/cerebral-debugger/package.json [hunk 8]
```diff
@@ -1,38 +1,60 @@
...
   },
+  "repository": "https://github.com/electron/electron-quick-start",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: wrong, keep previous

File: packages/cerebral-debugger/package.json [hunk 9]
```diff
@@ -1,38 +1,60 @@
...
   "keywords": [
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: review keywords

File: packages/cerebral-debugger/package.json [hunk 10]
```diff
@@ -1,38 +1,60 @@
...
-  "bugs": {
-    "url": "https://github.com/cerebral/cerebral/issues"
-  },
-  "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
-    "prepublish": "npm run build"
-  },
+  "author": "GitHub",
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: lie :)

File: packages/cerebral-debugger/package.json [hunk 11]
```diff
@@ -1,38 +1,60 @@
...
+    "express": "^4.14.0",
+    "react-scripts": "0.7.0",
+    "style-loader": "^0.13.1",
+    "url-loader": "^0.5.7",
+    "webpack": "^1.13.3",
+    "webpack-dev-server": "^1.16.2",
+    "webpack-hot-middleware": "^2.13.2",
+    "webpack-target-electron-renderer": "^0.4.0"
+  },
+  "babel": {
```
Comment (Guria) [inline packages/cerebral-debugger/package.json]: we are using .babelrc files across packages

File: package.json (hunk 1 above)
Comment (christianalfoni) [inline package.json]: Let me try, I think I met some issues last time... but yeah, will try

File: packages/cerebral-debugger/electron/LICENSE.md (hunk 2 above)
Comment (christianalfoni) [inline packages/cerebral-debugger/electron/LICENSE.md]: Copy paste, can remove it :)

File: packages/cerebral-debugger/package.json (hunk 3 above)
Comment (christianalfoni) [inline packages/cerebral-debugger/package.json]: Jup, will fix

File: packages/cerebral-debugger/package.json (hunk 7 above)
Comment (christianalfoni) [inline packages/cerebral-debugger/package.json]: Jup, have not looked into that cause I can not test it. So someone else has to look into it :)

File: package.json (hunk 1 above)
Comment (christianalfoni) [inline package.json]: It worked :)

**Comment by christianalfoni** [comment]
Okay, I did the fixes except:

- root `package.json` has to point to `electron/main.js` for `npm start` to work. It really is the main project there, cause that is what is used for development... so argue it makes sense that way ;-)

- Someone else has to look at windows/linux dists, cause I can not test here

The rest should be fixed :)

**Pull request merged**